quinn/src/endpoint.rs

```diff
@@ -244,12 +246,9 @@ impl Endpoint {
         let reason = Bytes::copy_from_slice(reason);
         let mut endpoint = self.inner.state.lock().unwrap();
         endpoint.connections.close = Some((error_code, reason.clone()));
-        for sender in endpoint.connections.senders.values() {
-            // Ignoring errors from dropped connections
-            let _ = sender.send(ConnectionEvent::Close {
-                error_code,
-                reason: reason.clone(),
-            });
+        for conn in endpoint.connections.refs.values() {
+            let mut state = conn.state.lock("close");
```

Comment:
Is the endpoint mutex always taken before the connection mutex in case both are used? This will be required to avoid deadlocks.

Reply:
When we're traversing connections via the endpoint, that's statically enforced, since the `ConnectionSet` is inside the `Endpoint` state mutex. Connection state is also locked independently via the public `Connection` API, but that shouldn't overlap with endpoint locks.

Things might get fishy if a caller captures and manipulates an `Endpoint` handle inside an extension trait object fed to that same endpoint, but I think even those will generally be invoked inside the endpoint mutex if any, and hence gracefully panic.